src/web/handlers.rs

```diff
@@ -352,40 +352,42 @@ pub fn post_collection_batch(
     )
 }
 
-pub fn delete_bso(bso_req: BsoRequest) -> impl Future<Output = Result<HttpResponse, Error>> {
+pub async fn delete_bso(bso_req: BsoRequest) -> Result<HttpResponse, Error> {
     bso_req.metrics.incr("request.delete_bso");
-    bso_req
+    match bso_req
         .db
         .delete_bso(params::DeleteBso {
             user_id: bso_req.user_id,
             collection: bso_req.collection,
             id: bso_req.bso,
         })
-        .map_err(From::from)
-        .map_ok(|result| HttpResponse::Ok().json(json!({ "modified": result })))
+        .await
```

Comment:
```suggestion
        .await?
```

Below is the old 404 code we fixed for #599. Instead it should just return the `HttpResponse::Ok` (when the ? above doesn't error out)

Reply:
The below needs to change to just return the `Ok` (and should fix the build)